src/components/AdaptiveCard.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { IconType } from 'react-icons/lib';
+import { z } from 'zod';
+
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+
+const VARIENT_ENUM = ['feature', 'course', 'blog'] as const;
+
+const BaseCardDataSchema = z.object({
+  title: z.string().optional(),
+  description: z.string().optional(),
+});
+
+const FeatureCardDataSchema = BaseCardDataSchema.extend({
+  icon: z.custom<IconType>().optional(),
+});
+
+const AdaptiveCardPropsSchema = z.object({
+  variant: z.enum(VARIENT_ENUM),
+  data: FeatureCardDataSchema,
+});
+
+type FeatureCardData = z.infer<typeof FeatureCardDataSchema>;
+type AdaptiveCardProps = z.infer<typeof AdaptiveCardPropsSchema>;
+
+export function AdaptiveCard({ variant, data }: AdaptiveCardProps) {
+  const result = AdaptiveCardPropsSchema.safeParse({ variant, data });
+
+  if (!result.success) {
+    console.log('🚀 ~ AdaptiveCard ~ result.error:', result.error.errors);
+  }
+
+  return (
+    <>{variant === 'feature' && <FeatureCardLayout {...result.data?.data} />}</>
```

Comment:
Unnecessary `<> </>` tags in return block, let's simplify it. It'll be eventually replaced with `if` blocks after implementing other card variants like `course, blog` etc.

For now, it could be: `return variant === 'feature' && <FeatureCardLayout {...result.data?.data} />`